backend/inspirehep/orcid/utils.py

```diff
@@ -226,3 +233,37 @@ def distributed_lock(lock_name, expire=10, auto_renewal=True, blocking=False):
 
 class DistributedLockError(Exception):
     pass
+
+
+def update_moved_orcid(old_orcid, new_orcid):
+    try:
+        author_record = AuthorsRecord.get_record_by_pid_value(old_orcid, "orcid")
+        if new_orcid not in get_value(author_record, "ids.value"):
+            new_author_ids = [
+                {"schema": "ORCID", "value": new_orcid},
+                *author_record["ids"],
+            ]
+            author_record["ids"] = new_author_ids
+            author_record.update(dict(author_record))
+    except PIDDoesNotExistError:
+        LOGGER.info(
+            "ORCID account was moved, but no profile matching ORCID number found on INSPIRE",
+            orcid=old_orcid,
+        )
+    user_orcid_account = RemoteAccount.query.filter(
+        RemoteAccount.user_id == UserIdentity.id_user, UserIdentity.id == old_orcid
+    ).one_or_none()
+    if user_orcid_account:
+        remove_access_token_for_orcid_account(user_orcid_account, old_orcid)
```

Comment:
there's something I hadn't thought of before: if we only delete the remote token, the user will still have a user account corresponding to that ORCID but without a valid token. However, if he tries to login again with his new ORCID, a new account will be created, but he won't be able to reuse the same email address. How should we deal with that @drjova? Deleting the whole user account seems like a better solution, but it might fail if the user has made any modifications to a record (e.g. jobs, seminars, author profile), as the SQLAlchemy transaction table has a foreign key on the user table. The whole account handling is kind of a mess currently.